### ISSUE TITLE
Request: Add metadata cache for jSDelivr and Unpkg providers

### ISSUE THREAD

**Issue by RobJohnston** [aspnet/LibraryManager#598]
Based on https://github.com/aspnet/LibraryManager/issues/370#issuecomment-637953775 and https://github.com/aspnet/LibraryManager/issues/389#issuecomment-579920081, the library metadata for the Unpkg and JSDelivr providers is not yet cached.  This causes libman to make some requests on every restore using those providers, which should be unnecessary.  Addressing this may resolve some speed and reliability issues.

Is anyone currently working on this or is it up for grabs?

**Comment by jimmylewis** [comment]
@RobJohnston I'm going to try to squeeze this in next week if I can, but feel free if you want to try it before then.

I think the cache behavior for the CDNJS provider also needs to be reviewed, in case it's throwing out the local cached copy even when a refresh request fails.  Due to how that caching works (IIRC all versions of a given library are in a single metadata file/request) we can't just rely on a local copy exclusively, we do need to try and refresh in case a newer version of the library has been released.

**Comment by RobJohnston** [comment]
I only started to feel my way through the problem with jsDelivr using CDNJS as an example... haven't gotten too deep into it.

You are correct that all versions are in a single metadata file for CDNJS.  For example, https://api.cdnjs.com/libraries/jquery.  One challenge is that the other providers don't have such an endpoint in their APIs.  

jsDelivr used to have a similar endpoint, but the advice with their new API is to query all the versions and then get the version that you want (see https://github.com/jsdelivr/data.jsdelivr.com/issues/7).  What UNPKG can provide is apparently still up for discussion (see https://github.com/mjackson/unpkg/issues/85).  

I was thinking ahead when I made this issue... I thought the first step would be to get the caching done and then work on a resiliency strategy in a follow-up issue (likely with the cache being the fallback when all else fails).
